test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterFactory.java

```diff
@@ -196,11 +196,15 @@ public String getTransportEndpoint() {
         }
 
         public String getRemoteClusterServerEndpoint() {
-            Path portsFile = workingDir.resolve("logs").resolve("remote_cluster.ports");
-            if (Files.notExists(portsFile)) {
-                waitUntilReady();
+            if (spec.isRemoteClusterServerEnabled()) {
+                Path portsFile = workingDir.resolve("logs").resolve("remote_cluster.ports");
+                if (Files.notExists(portsFile)) {
+                    waitUntilReady();
+                }
+                return readPortsFile(portsFile).get(0);
+            } else {
+                return "";
```

Comment:
Unlike HTTP and Transport, different nodes can have different values for whether remote cluster server is enabled. So we need a way to not throw exception by checking whether server mode is actually enabled.

Reply:
> this wasn't an issue before because we simply weren't calling it for mixed-node clusters

We didn't have a cluster that has `remote_cluster_server` partially enabled, i.e. some nodes enable it while some don't. This is also an unique situation to the remote cluster server port compared to Transport and HTTP ports which must always be enabled on all nodes.

> Actually, looking at waitUntilReady, we're not actually waiting for the remote_cluster.ports

Err you are right. I can tackle it in a follow-up.